src/wp-includes/l10n.php

```diff
@@ -788,22 +788,81 @@ function load_textdomain( $domain, $mofile, $locale = null ) {
 		$locale = determine_locale();
 	}
 
-	$mo = new MO();
-	if ( ! $mo->import_from_file( $mofile ) ) {
-		$wp_textdomain_registry->set( $domain, $locale, false );
+	// Ensures the correct locale is set as the current one, in case it was filtered.
+	WP_I18n_Translation_Controller::instance()->set_locale( $locale );
```

Comment:
Nit-pick: Since the controller is accessed several times in this function, it would be a bit more readable to assign `WP_I18n_Translation_Controller::instance()` to a variable.